runtime/src/main/java/org/corfudb/protocols/logprotocol/SMRLogEntry.java

```diff
@@ -0,0 +1,137 @@
+package org.corfudb.protocols.logprotocol;
+
+import io.netty.buffer.ByteBuf;
+import lombok.Getter;
+import lombok.ToString;
+import lombok.extern.slf4j.Slf4j;
+import org.corfudb.runtime.CorfuRuntime;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+
+/**
+ * A log entry structure which contains a collection of multiSMRentries,
+ * each one contains a list of updates for one object.
+ */
+@SuppressWarnings("checkstyle:abbreviation")
+@ToString
+@Slf4j
+public class SMRLogEntry extends LogEntry {
+
+    // Map from stream-ID to a list of SMR updates to this stream.
+    public Map<UUID, List<SMRRecord>> entryMap = Collections.synchronizedMap(new HashMap<>());
+
+    public SMRLogEntry() {
+        this.type = LogEntryType.SMRLOG;
+    }
+
+    public SMRLogEntry(Map<UUID, List<SMRRecord>> entryMap) {
+        this.type = LogEntryType.SMRLOG;
+        this.entryMap = entryMap;
+    }
+
+    /**
+     * Extract a particular stream's entry from this object.
+     *
+     * @param streamID stream ID
+     * @return the SMR Record list corresponding to streamId
+     */
+    private List<SMRRecord> getStreamEntry(UUID streamID) {
+        return entryMap.computeIfAbsent(streamID, sid -> new ArrayList<>());
+    }
+
+    /**
+     * Add one SMR-update to one object's update-list.
+     *
+     * @param streamID  StreamID
+     * @param smrRecord SMRRecord to add
+     */
+    public void addTo(UUID streamID, SMRRecord smrRecord) {
+        getStreamEntry(streamID).add(smrRecord);
+    }
+
+    /**
+     * Merge two SMRLogEntry records.
+     * Merging is done object-by-object.
+     *
+     * @param other object to merge
+     */
+    public void mergeInto(SMRLogEntry other) {
+        if (other == null) {
+            return;
+        }
+
+        other.getSMRUpdates().forEach((streamID, smrRecords) -> {
+            getStreamEntry(streamID).addAll(smrRecords);
+        });
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use explicit scoping instead of the default package private level](https://app.codacy.com/app/annymartinez.m/CorfuDB/pullRequest?prid=4651468)